aisdc/attacks/worst_case_attack.py

```diff
@@ -426,23 +468,79 @@ def _construct_metadata(self):
         # Global metrics
         self.metadata["global_metrics"] = self._get_global_metrics(self.attack_metrics)
         self.metadata["baseline_global_metrics"] = self._get_global_metrics(
-            self.dummy_attack_metrics
+            self._unpack_dummy_attack_metrics_experiments_instances()
         )
 
+    def _unpack_dummy_attack_metrics_experiments_instances(self) -> list:
+        """Constructs the metadata object, after attacks"""
+        dummy_attack_metrics_instances = []
+
+        for exp_rep in range(len(self.dummy_attack_metrics)):
+            temp_dummy_attack_metrics = self.dummy_attack_metrics[exp_rep]
+            dummy_attack_metrics_instances += temp_dummy_attack_metrics
+
+        return dummy_attack_metrics_instances
+
+    def _get_attack_metrics_instances(self) -> dict:
+        """Constructs the metadata object, after attacks"""
+        attack_metrics_experiment = {}
+        attack_metrics_instances = {}
+
+        for rep in range(len(self.attack_metrics)):
+            attack_metrics_instances["instance_" + str(rep + 1)] = self.attack_metrics[
+                rep
+            ]
+
+        attack_metrics_experiment["attack_instance_logger"] = attack_metrics_instances
+        attack_metrics_experiment[
+            "attack_metric_failfast_summary"
+        ] = self.attack_metric_failfast_summary.get_attack_summary()
+
+        return attack_metrics_experiment
+
+    def _get_dummy_attack_metrics_experiments_instances(self) -> dict:
+        """Constructs the metadata object, after attacks"""
+        dummy_attack_metrics_experiments = {}
+
+        for exp_rep in range(len(self.dummy_attack_metrics)):
+            temp_dummy_attack_metrics = self.dummy_attack_metrics[exp_rep]
+            dummy_attack_metric_instances = {}
+            for rep in range(len(temp_dummy_attack_metrics)):
+                dummy_attack_metric_instances[
+                    "instance_" + str(rep + 1)
+                ] = temp_dummy_attack_metrics[rep]
+            temp = {}
+            temp["attack_instance_logger"] = dummy_attack_metric_instances
+            temp[
+                "attack_metric_failfast_summary"
```

Comment:
Would it be neater to not sumamrise here? Why not put in all the failfast stuff and summarise later? Or is this just a description to help describe why the experiment stopped?